paasta_tools/paasta_metastatus.py

```diff
@@ -162,20 +178,34 @@ def assert_quorum_size(state):
                 False)
 
 
-def get_mesos_status():
+def assert_extra_slave_data(mesos_state):
+    header_string = '%10s %36s %9s' % ('Hostname', 'CPU free', 'RAM free')
+    slave_outputs = ['%40s %8.2f %9.2f' % (
+        slave['hostname'],
+        slave['free_resources']['cpus'],
+        slave['free_resources']['mem'],
+    ) for slave in get_extra_mesos_slave_data(mesos_state)]
+    check_output = [header_string] + slave_outputs
+    return ('\n'.join(check_output), True)
+
+
+def get_mesos_status(mesos_state, verbosity):
     """Gathers information about the mesos cluster.
        :return: tuple of a string containing the status and a bool representing if it is ok or not
     """
 
-    state = get_mesos_state_from_leader()
-    cluster_results = run_healthchecks_with_param(state, [assert_quorum_size, assert_no_duplicate_frameworks])
+    cluster_results = run_healthchecks_with_param(mesos_state, [assert_quorum_size, assert_no_duplicate_frameworks])
 
     metrics = get_mesos_stats()
     metrics_results = run_healthchecks_with_param(metrics, [
         assert_cpu_health,
         assert_memory_health,
+        assert_tasks_running,
         assert_slave_health,
-        assert_tasks_running])
+    ])
+
+    if verbosity >= 2:
+        metrics_results.extend(run_healthchecks_with_param(mesos_state, [assert_extra_slave_data]))
```

Comment:
I think it'd be better if you went with what I suggested originally: move the array of checks into a variable, append to it if the verbose flag is set >=2, and then pass that array to ``run_healthchecks_with_param``. It's not a huge deal, just means ``run_healthchecks_with_param`` is only called once.